### PR TITLE
return no-op span builder after shutdown

### PR DESCRIPTION
**Pull request by brettmc**
this completes a todo in code comments: when tracer shared state has shutdown, getting a span builder should return a no-op implementation

### CODE REVIEW

**Comment by brettmc** [comment]
NB that I needed to adjust a bunch of tests, as tracer shared state was immediately shutting down due to their not maintaining a reference to the tracer provider. Should be fixed by #760

File: src/SDK/Trace/Tracer.php [hunk 1]
```diff
@@ -34,7 +35,7 @@ public function spanBuilder(string $spanName): API\SpanBuilderInterface
         }
 
         if ($this->tracerSharedState->hasShutdown()) {
-            // TODO: Return a noop tracer
+            return new API\NoopSpanBuilder(Context::storage());
```
Comment (tidal) [inline src/SDK/Trace/Tracer.php:38]: Can you add this to the class' unit tests?

Comment (brettmc) [reply src/SDK/Trace/Tracer.php:38]: done

**Review by tidal — APPROVED** [review]

**Comment by tidal** [comment]
There is a huge drop in code-coverage, which is unrelated. However [SpanTest](https://github.com/open-telemetry/opentelemetry-php/blob/main/tests/Unit/SDK/Trace/SpanTest.php) has only a @coversDefaultClass annotation, which is never used, so all those tests are completely ignored by code-coverage.

**Pull request merged**